### ISSUE TITLE
Suggestion for server profiles

### ISSUE THREAD

**Issue by flareofghast** [hexparrot/mineos-node#29]
I'm not sure what you desire to do with the profile page but I think that the page would look better if perhaps the available downloads are in a dropdown list and the installed profiles are in a list on the page. Each type of server in it's own DDL e.g. Vanilla, FTB etc. If there was a method to check if the vanilla server jar is already downloaded and copy or symlink in to the new directory instead of downloading.

**Comment by hexparrot** [comment]
Thanks to this discourse post, I was able to fully implement profiles.  http://discourse.codeemo.com/t/help-needed-regarding-server-pack-downloads/505/5?u=hexparrot

Thing went smoothly with the commits I've written over the past few days so I've actually implemented it into the master branch, and it's operating as I've hoped!

It shouldn't be too hard to replace the existing filter button with two drop downs, one for each 'group' (mojang,ftb,ftb_third_party) as well as for (release|snapshot|old_version).

It's a good idea! I like it, but its mostly aesthetic, so right now I want to keep chugging away at all the core functionality so if you'd like to give it a spin, I welcome it!

**Comment by flareofghast** [comment]
I noticed you had been quite busy.

**Comment by flareofghast** [comment]
Did you also make the recommended/provided start options / arguments propagate the server.config?

**Comment by hexparrot** [comment]
@flareofghast what do you mean re: start options/arguments?

**Comment by flareofghast** [comment]
when you download one of their releases it comes with a start.sh (or something like that) which has xms values etc, thought that perhaps there is a reason they set those.

**Comment by flareofghast** [comment]
do you stay up all night coding some times?

**Comment by flareofghast** [comment]
in serverStart.sh

java -server -Xms512m -Xmx2048M -XX:PermSize=256m -d64 -XX:+UseParNewGC -XX:+CMSIncrementalPacing -XX:+CMSClassUnloadingEnabled -XX:ParallelGCThreads=2 -XX:MinHeapFreeRatio=5 -XX:MaxHeapFreeRatio=10 -jar FTBServer-1.7.10-1352.jar nogui

**Comment by kalvindukes** [comment]
Whoa! Another great thing done, Will! :D
Did you get my emails (2) today?
@flareofghast Some of the FTBServers (I know it in Infinity Pack for sure) come with an install.sh which has to be run before starting the server for the first time. But the stuff in start.sh seems to be arbitrary... My server wouldn't even start with PermSize 256m ;)

**Comment by flareofghast** [comment]
the install.sh makes sure to have the correct official minecraft jar and launch wrapper available

**Comment by flareofghast** [comment]
I assume that hexparrot tests the downloads and runs them to make sure they work before pulling in the branch

**Comment by kalvindukes** [comment]
I also just realized that there's just PermSize set, not MaxPermSize...

**Comment by hexparrot** [comment]
````java -server -Xms512m -Xmx2048M -XX:PermSize=256m -d64 -XX:+UseParNewGC -XX:+CMSIncrementalPacing -XX:+CMSClassUnloadingEnabled -XX:ParallelGCThreads=2 -XX:MinHeapFreeRatio=5 -XX:MaxHeapFreeRatio=10 -jar FTBServer-1.7.10-1352.jar nogui````

I'll never aim to include arguments like these into the java startup string--that will always be up to the admin himself.  I have a long history of strongly opposing such startup scripts as seen from this lengthly flamewar of a bukkit post: http://bukkit.org/threads/optimize-java-for-minecraft-in-a-extreme-way-p-need-feedback-thx-i-am-back.25994/page-2#post-487416

That and startup scripts never start the server with `screen`, which is a necessity for MineOS....

@flareofghast I don't really test the packs--or at least, not in any systematic way.  As far as I'm concerned, I need to program a webui that runs a jarfile.  When I can do some additional coding to make the webui more usable, I *want* to do it, but not to the extent that I am hard-coding pack-specific instructions (that will inevitably go obsolete).

I hope I can produce code that can be sweepingly useful to all packs...just making sure to cover all the edge cases, such as dependencies of Mojang jars,  etc

**Comment by flareofghast** [comment]
see new issue, thought might be easier to track things if I have an issue per... issue.
